src/transformers/tokenization_utils.py

```diff
@@ -408,52 +448,52 @@ def _add_tokens(self, new_tokens: Union[List[str], List[AddedToken]], special_to
         # Note: resize_token_embeddings expects to receive the full size of the new vocabulary, i.e. the length of the tokenizer.
         model.resize_token_embeddings(len(tokenizer))
         ```"""
-        new_tokens = [str(tok) for tok in new_tokens]
-
-        tokens_to_add = []
+        added_tokens = 0
+        if new_tokens is None:
+            return added_tokens
+        current_vocab = self.get_vocab().copy()
+        new_idx = len(set(current_vocab.keys()))  # only call this once, len gives the last index + 1
```

Comment:
I don't think you need to `set` or `keys` call here - keys are already unique

```suggestion
        new_idx = len(current_vocab)  # only call this once, len gives the last index + 1
```